monkey/tests/unit_tests/infection_monkey/island_api_client/test_http_client.py

```diff
@@ -0,0 +1,119 @@
+from unittest.mock import MagicMock
```

Comment:
I don't see any tests for `POST`.

Reply:
Just test that the requests mock gets called with the expected JSON

Reply:
You can probably parametrize all `put` tests to test both `put` and `post`

Reply:
The readability of that unit test would be pretty bad as you would have to understand the unorthodox calls of a method by name. These unit tests are already bringing a lot of code to basically test non-existent logic. If the unit test contains 10 times more logic than the actual code, then it's probably not worth writing it, as you're 10 times more likely to leave a bug in the unit test.